### ISSUE TITLE
[PRE REVIEW]: PyVOLCANS: A Python package to flexibly explore similarities and differences between volcanic systems

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3436]
**Submitting author:** @PTierz (<a href="http://orcid.org/0000-0001-8889-9900">Pablo Tierz Lopez</a>)
**Repository:** <a href="https://github.com/BritishGeologicalSurvey/pyvolcans" target ="_blank">https://github.com/BritishGeologicalSurvey/pyvolcans</a>
**Version:** v1.1.2
**Editor:** @elbeejay
**Reviewers:** @jifarquharson, @meghanrjones
**Managing EiC:** Arfon Smith

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/5148a124aa6c9aecf5936022dd1a2bf3/status.svg)](https://joss.theoj.org/papers/5148a124aa6c9aecf5936022dd1a2bf3)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/5148a124aa6c9aecf5936022dd1a2bf3"><img src="https://joss.theoj.org/papers/5148a124aa6c9aecf5936022dd1a2bf3/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/5148a124aa6c9aecf5936022dd1a2bf3/status.svg)](https://joss.theoj.org/papers/5148a124aa6c9aecf5936022dd1a2bf3)
```

**Author instructions**

Thanks for submitting your paper to JOSS @PTierz. **Currently, there isn't an JOSS editor assigned** to your paper.

@PTierz if you have any suggestions for potential reviewers then please mention them here in this thread (without tagging them with an @). In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission (please start at the bottom of the list).

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.04 s (263.5 files/s, 106419.3 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                           8            640           1605           1714
Markdown                         1             99              0            324
YAML                             2              9              0             52
-------------------------------------------------------------------------------
SUM:                            11            748           1605           2090
-------------------------------------------------------------------------------


Statistical information for the repository 'f8beae631ff112e973e4138a' was
gathered on 2021/06/30.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
John A Stevenson                17          2649            186           41.44
PTierz                         159          1897            729           38.39
Tierz Lopez                     10           192             54            3.60
mobiuscreek                     22           405            276            9.95
ptierz                           1             1              1            0.03
vyronc                          30           251            200            6.59

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
John A Stevenson           2560           96.6          7.5               15.27
PTierz                     1158           61.0          3.3               11.23
mobiuscreek                  43           10.6          0.6                0.00
vyronc                      198           78.9          5.7                1.52
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3436 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by arfon** [comment]
@whedon generate pdf from branch JOSS_paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch JOSS_paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03436/joss.03436/10.21105.joss.03436.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03436/joss.03436/10.21105.joss.03436.pdf) :page_facing_up: :point_left:

**Comment by arfon** [comment]
@whedon invite @elbeejay as editor

:wave: @elbeejay – would you be willing to edit this submission for us?

**Comment by whedon** [comment]
@elbeejay has been invited to edit this submission.

**Comment by elbeejay** [comment]
@whedon assign me as editor

**Comment by whedon** [comment]
OK, the editor is @elbeejay

**Comment by elbeejay** [comment]
Hello @PTierz, thanks for your submission to JOSS. I applaud you and your co-authors on porting your MATLAB code to Python in an effort to make VOLCANS fully open-access. I have a few questions during this pre-review stage of the JOSS process:

1. In the project `README.md` it is [mentioned](https://github.com/BritishGeologicalSurvey/pyvolcans#analogy-matrices) that there are plans to convert the calculation of analogy matrices to Python - what is the anticipated timeline for this addition? I think adding this component to the code base would make this submission stronger and the methodology more transparent, by making it easier for a user to modify and/or extend the code if needed.
2. Beyond the information provided in the project `README.md` I have been unable to find further documentation (e.g. API documentation, community guidelines). Can you direct me to where I might find this information? The expected documentation for JOSS submissions can be found [here](https://joss.readthedocs.io/en/latest/review_criteria.html#documentation). 

Thank you, let me know if any of these questions require further clarification.

**Comment by PTierz** [comment]
Hello @elbeejay,

Very many thanks for your message and your encouraging words. We sincerely believe that `PyVOLCANS` [v1.1.2](https://github.com/BritishGeologicalSurvey/pyvolcans/tree/v1.1.2) can already represent a very important step forward for our community.

Unfortunately, the second half of this week has been quite busy, and I will not be able to provide you with substantial feedback on your two queries by the end of today. Please accept my apologies.

I am currently in discussions with the co-authors to be able to send you appropriate, full responses to your questions over the following 1-2 weeks, OK?

Thank you very much in advance for your support.

Best regards,
Pablo

**Comment by elbeejay** [comment]
> I am currently in discussions with the co-authors to be able to send you appropriate, full responses to your questions over the following 1-2 weeks, OK?

Great, this is no problem, please take as much time as you need.

**Comment by PTierz** [comment]
Dear @elbeejay,

Very many thanks for your support and your patience. Your comments have been very useful for us to improve the documentation for our program, and to revisit the timeline and requirements needed to successfully continue with our 'translation' of the Matlab VOLCANS code, into the Python PyVOLCANS code. Many thanks.

Concerning the first point you raised, there are indeed plans of converting the entire set of Matlab scripts that calculate the volcano analogies into Python, and release them in future versions of the PyVOLCANS code. We sincerely agree with the fact that this would be a desirable addition to the program, as we acknowledge in our recently created [wiki pages](https://github.com/BritishGeologicalSurvey/pyvolcans/wiki). However, due to the reasons explained below, we truly believe that the current submission of PyVOLCANS v1.1.2 has a substantial scientific value on its own, hence making it a suitable contribution to JOSS. Many thanks.

The timeline for the full Matlab-Python conversion might be significantly long (perhaps 1-2 years from now). This timeline would also depend on the availability of internal funding within our organisation, which should support the required team programming effort.

Nevertheless, we are always open to share further details of the VOLCANS methodology with any interested user, while we are translating the entire code into Python language, as it is now explicitly written in our [wiki pages](https://github.com/BritishGeologicalSurvey/pyvolcans/wiki).
In a similar way, we were always completely open to the several requests we received from colleagues to compute sets of analogue volcanoes using VOLCANS, from December 2019 (when the VOLCANS work was first presented to the scientific community) to March 2021, when the first publicly available version of PyVOLCANS was released.

Since then, we have received significant [attention](https://twitter.com/volcan01010/status/1384108463505117197) and [usage](https://github.com/BritishGeologicalSurvey/pyvolcans/graphs/traffic) from a variety of potential users, which clearly supports our case to defend the relevance of the submitted PyVOLCANS v1.1.2 code. We wholeheartedly believe that making the VOLCANS results openly available to any volcano scientist and researcher in the world, holds a significant value for the scientific community (and the public). This is what the current PyVOLCANS code does, making it a working Minimum Viable Product, with reusable and testable code that enables repeatable and reproducible research. In our opinion, this stands as a very valuable first step of (ongoing) development, which we carried out, as a team, over the last year. We would also like to stress that PyVOLCANS v1.1.2 represents an enabling technology, for it allows users to carry out a great variety of scientific analyses using data-driven sets of analogue volcanoes, something that was definitely not possible before the release of the code. Very many thanks.

Concerning your second point, in order to tackle the issues you observed, we have created a series of [wiki pages](https://github.com/BritishGeologicalSurvey/pyvolcans/wiki) inside our repository, to clearly provide every user of PyVOLCANS with detailed information on how to install, use and modify our code.

In particular, in the [README.md](https://github.com/BritishGeologicalSurvey/pyvolcans#readme) file, we have clearly described the following aspects: (1) statement of need and target audience of PyVOLCANS; (2) installation instructions, following PyPI for standard use or through cloning the GitHub repository and running our [setup.py](https://github.com/BritishGeologicalSurvey/pyvolcans/blob/main/setup.py) file for development use of the code; (3) example usage (which is significantly expanded in the [wiki pages](https://github.com/BritishGeologicalSurvey/pyvolcans/wiki)); (4) API documentation (we also point our future users to the source scripts and the [wiki pages](https://github.com/BritishGeologicalSurvey/pyvolcans/wiki) to find more details on all the functions used in the code and how to run different types of analyses); (5) community guidelines, including how to contribute to the code, report issues/bugs and seek support (this information is also detailed in the [wiki pages](https://github.com/BritishGeologicalSurvey/pyvolcans/wiki)).

We most sincerely hope that all the aforementioned additions to PyVOLCANS, together with our responses, can help our manuscript to go through the next steps in the peer-review process.

If you required anything else from us, please do not hesitate to let us know (please note that I, Pablo, will be taking summer annual leave from 24th July to 16th August. Therefore, it is unlikely that I will be able to successfully proceed with my review tasks during those weeks. Very many thanks in advance for your understanding).

Best regards,
Pablo (on behalf of the authors)

**Comment by elbeejay** [comment]
Hi @PTierz, thank you for this explanation and thoughtful response.

I do not doubt the usefulness of the software for the community, however I still struggle slightly with the submitted software lacking the foundational computations which create the analog matrices, as well as the lack of [API documentation](https://joss.readthedocs.io/en/latest/review_criteria.html#api-documentation).

Given your leave coming up, and these outstanding doubts I have, I will be sharing this submission with the broader JOSS editorial team to get their opinions on whether or not the software and its documentation are fit for JOSS at this time, or whether the full analog matrix computations and full API documentation are needed for this submission to fit within the JOSS scope.

We will likely reach a decision by the end of next week, but given your upcoming leave, I will "pause" this pre-review and we can resume the process on August 16th.

**Comment by elbeejay** [comment]
@whedon query scope

**Comment by whedon** [comment]
Submission flagged for editorial review.

**Comment by volcan01010** [comment]
Hi @elbeejay, thank you for reviewing our submission,

Since the original submission we have restructured our README.md and created the [wiki](https://github.com/BritishGeologicalSurvey/pyvolcans/wiki) page.

The aim of PyVOLCANS is to allow relatively non-technical users the query the analogy matrix data files, so the only real user-facing API is the command-line interface.  The description of this is now more prominent on the [README.md](https://github.com/BritishGeologicalSurvey/pyvolcans/tree/readthedocs#readme).  More complex examples are now in the wiki, where they are better-structured than they were on the single-page README.md.

The [review criteria](https://joss.readthedocs.io/en/latest/review_criteria.html#api-documentation) say that functions/methods are documented.  While we do not have a ReadTheDocs page, or similar, they all have full Numpy-style doc strings.

Is the updated documentation suitable now?  If not, could you point to some specific examples of what you think it need, please?

**Comment by danielskatz** [comment]
Just to interject as one of the JOSS AEiCs, the JOSS concern about API documentation is related to the fact that this is open source software where we hope that others developers will contribute to and use the software or elements of it, not just as end users.  So please don't think of our API documentation review criterion as entirely user-facing.

**Comment by PTierz** [comment]
Hi @elbeejay, 

Very many thanks for your further comments. I just would like to comment on the topics concerning the validity, usefulness, and transparency of VOLCANS and, therefore, PyVOLCANS. Many thanks in advance.

Unfortunately, it is a bit difficult for us to fully understand where our submitted manuscript fails in terms of complying with the [submission requirements]( https://joss.readthedocs.io/en/latest/submitting.html), or the scope, of JOSS. We sincerely believe that one of the greatest strengths of our submission lies on the requirement: “The software must have an **obvious** research application.” We think that this is clearly demonstrated in the [second last paragraph of our submission](https://github.com/openjournals/joss-papers/blob/joss.03436/joss.03436/10.21105.joss.03436.pdf), where up to four scientific studies over 2020-2021 have been supported by the knowledge at the base of PyVOLCANS. In early 2020, we had to make a decision between two options: either focusing on the full translation of VOLCANS from Matlab to Python, which would have required 2-3 years of development (being rather optimistic); or focusing, as an initial stage of development, on making the VOLCANS results openly available to everyone, so people could benefit from this first-of-its-kind, enabling technology to tackle issues related to data scarcity in volcanology.

If we had chosen the first route, the obvious research application of PyVOLCANS would have been kept away from the scientific community during the several years that we would have needed to develop a PyVOLCANS version that translated the entire VOLCANS method into Python. This is not to mention the enhanced risks of such a long-term project, with strong dependence on an ever-changing funding landscape, both internally in our institution, and externally, from national and/or international funding streams. In other words, chances were that the complete version of PyVOLCANS would have never been created.

Therefore, we struggle to see how the current version of PyVOLCANS is not deemed as a valuable contribution to research in volcanology, and perhaps beyond, as volcanic systems have played and play many important roles in the evolution and functioning of our geodynamic planet.

We would also like to remark that the datasets that support the creation of the analogy matrices were made fully open-access in 2019, as [Electronic Supplementary Material of the VOLCANS article]( https://link.springer.com/article/10.1007/s00445-019-1336-3#Sec25). Complementarily, the [methodology behind the calculation of the analogy matrices]( https://link.springer.com/article/10.1007/s00445-019-1336-3#Sec6) is carefully and thoroughly detailed in the aforementioned article, and it went through a comprehensive peer-review process carried out by knowledgeable scientific colleagues.

In summary, we are committed to full transparency about how PyVOLCANS derives the sets of analogue volcanoes that anyone in the world can now calculate via the Python code that we submitted for publication in JOSS. We sincerely consider that this submitted version already represents a significant step forward: i.e. a minimum viable product with obvious research applications.

Very many thanks again for your feedback and encouragement to improve our research and code development.

We will look forward to hearing from you and the editorial team of JOSS whenever it may be possible.

Receive our best regards,
Pablo (on behalf of the authors)

**Comment by elbeejay** [comment]
Hi @PTierz, I hope you had a good holiday. I've conferred with the JOSS editorial team and am happy to say we will be moving PyVOLCANS on to the review stage. 

The one recommendation we have is that the MATLAB code for generating the analog matrices should be added to the repository. This would enable an outside contributor to see the code behind that computation, or potentially perform that Matlab-Python conversion for you! Hopefully this strikes a reasonable compromise between making the available code as transparent and "open" as possible, without placing the burden of doing what sounds like an arduous conversion of code on the development team.

Can you recommend any reviewers that would be a good fit to review this work? A list of folks that have expressed a willingness to review for JOSS is available [here](https://bit.ly/joss-reviewers), but feel free to also recommend people not on this list. Please do not tag potential reviewers in this thread but rather provide their GitHub handle without the "@" symbol (e.g. "elbeejay" for me).

Thanks, let me know if you have any outstanding questions about the process.

**Comment by PTierz** [comment]
Hi @elbeejay,

Many thanks for your message and your wishes (holidays were great, thanks). Apologies for the delay in replying.

It is very positive news to hear that our submission of `PyVOLCANS` can continue into the review stage. I thank you and the JOSS editorial team for your work and support.

We are happy to comply with your recommendation and upload the Matlab scripts that I used to generate the analogy matrices. Your point about outside contributors being able to see the Matlab code and collaborate to its conversion into Python is an important one. Many thanks for your feedback. We sincerely hope that, through collaboration with the scientific community, `PyVOLCANS` can become an increasingly useful tool for volcanologists and other scientists.

I am glad to inform you that I have just uploaded the relevant [set of Matlab scripts](https://github.com/BritishGeologicalSurvey/pyvolcans/tree/matlab-scripts/pyvolcans/VOLCANS_matlab_scripts), together with a [README_VOLCANS.md file](https://github.com/BritishGeologicalSurvey/pyvolcans/blob/matlab-scripts/pyvolcans/VOLCANS_matlab_scripts/README_VOLCANS.md) that documents some extra details on `VOLCANS` and on the concept of _volcano analogy_, and provides a brief description of the main purpose of each of the scripts uploaded. It also contains details of what data are stored in each of the csv files that are used by the scripts (please find additional descriptions of the data contained in the csv files in the [README_NUMBERING.md file](https://github.com/BritishGeologicalSurvey/pyvolcans/blob/matlab-scripts/pyvolcans/VOLCANS_csv_files/VOLCANS_code_numbers/README_NUMBERING.md) that we have also created. Many thanks).

We most sincerely hope that these additions can be satisfactory. If you required anything else from us, please do not hesitate to let us know. Many thanks.

Finally, in terms of reviewers, we consider that the following three could be a good fit for our manuscript: jifarquharson, weiji14, rbeucher.

I believe that we do not have any specific questions at this time, but many thanks for your availability.

Receive our best regards,

Pablo (on behalf of the authors)

**Comment by elbeejay** [comment]
@PTierz great! I think this works well and gives the interested user/reader an opportunity to both see the underlying code and contribute to the overall project if they so desire. Thank you for being so accommodating.

**Comment by elbeejay** [comment]
:wave: @jifarquharson, @weiji14, @rbeucher, would you be willing to review this submission to JOSS? We carry out our checklist-driven reviews here in GitHub issues and follow these guidelines: [https://joss.readthedocs.io/en/latest/review_criteria.html](https://joss.readthedocs.io/en/latest/review_criteria.html)

The submission is "PyVOLCANS: A Python package to flexibly explore similarities and differences between volcanic systems". I am reaching out to you as you all are geoscientists with experience coding and working in Python.

This is a pre-review issue which is used to find reviewers. Once 2 or 3 reviewers have been found I'll start the review on a dedicated GitHub issue. At present we are asking reviewers to complete reviews in 6 weeks, although this can be extended if needed. If you are not able to review but can recommend someone else, please mention them here (in this case please mention their GitHub handle without the "@" symbol).

If you are interested in reviewing, please take a look at the journal's [conflict of interest](https://joss.readthedocs.io/en/latest/reviewer_guidelines.html?#joss-conflict-of-interest-policy) policy to ensure you do not have a conflict before agreeing to review this submission.

Thank you, feel free to reach out if you have any questions about the JOSS review process. Please do not feel any pressure to accept this review request if you do not have the time or do not feel comfortable reviewing this software package, we appreciate and respect our peer-reviewers' time.

-Jay

**Comment by jifarquharson** [comment]
Hi @elbeejay, I'd be happy to review this submission.
- Jamie

**Comment by elbeejay** [comment]
@whedon add @jifarquharson as reviewer

**Comment by whedon** [comment]
OK, @jifarquharson is now a reviewer

**Comment by rbeucher** [comment]
Hi, yes I should be able to help.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by weiji14** [comment]
Hi @elbeejay, I won't be able to review this submission, but I can recommend @meghanrjones who would be more suitable.

**Comment by rbeucher** [comment]
Note that I don't have much expertise on volcanoes. So if @meghanrjones is happy to review...

**Comment by meghanrjones** [comment]
@elbeejay, I could review this submission. I do not have prior experience with JOSS but have expertise in OSS and volcanology.

**Comment by elbeejay** [comment]
That would be fantastic, thank you @meghanrjones! In that case, thanks also to @weiji14 and @rbeucher, but it seems your services will not be needed on this one.

**Comment by elbeejay** [comment]
@whedon add @meghanrjones as reviewer

**Comment by whedon** [comment]
OK, @meghanrjones is now a reviewer

**Comment by elbeejay** [comment]
@whedon start review

**Comment by whedon** [comment]
OK, I've started the review over in https://github.com/openjournals/joss-reviews/issues/3649.
